cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/json"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"path"
+	"strconv"
+	"strings"
+)
+
+func npmExecuteLint(config npmExecuteLintOptions, telemetryData *telemetry.CustomData) {
+	utils := npm.NpmUtilsBundle{}
+
+	err := runNpmExecuteLint(&utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runNpmExecuteLint(utils npm.NpmUtils) error {
+	packageJSONFiles, err := npm.FindPackageJSONFiles(utils)
+	if err != nil {
+		return err
+	}
+
+	packagesWithCilint, err := findPackagesWithCilint(packageJSONFiles, utils)
+	if err != nil {
+		return err
+	}
+
+	if len(packagesWithCilint) > 0 {
+		err = runCiLint(packagesWithCilint, utils)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = runDefaultLint(utils)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func findPackagesWithCilint(packageJSONFiles []string, utils npm.NpmUtils) ([]string, error) {
+	var packageJSON map[string]interface{}
+	var packagesWithCilint []string
+
+	for _, file := range packageJSONFiles {
+		packageRaw, err := utils.FileRead(file)
+		if err != nil {
+			return nil, err
+		}
+
+		json.Unmarshal(packageRaw, &packageJSON)
```

Comment:
Unmarshal() returns an error. I guess for what you do below, you could prepend `_ = ` to ignore the error explicitly.
I am not sure of the side-effects of re-using the `packageJSON` in each loop iteration. I wouldn't even spend the effort to check whether Unmarshal() will clear the map first, just declare the variable inside the loop.

Reply:
Yeah that was a problem, that Unmarshal() does not clear the map, but yes declaring the variable inside the loop is better then.